tests/Unit/JobApplicationTest.php

```diff
@@ -265,8 +267,123 @@ public function testSaveProfileSnapshotResetEachTime(): void
+     * and that it preserves relationships between copies.
+     *
+     * @return void
+    */
+    public function testSaveProfileSnapshotTimeline(): void
+    {
+        $applicant = factory(Applicant::class)->create();
+        $experience = factory(ExperienceAward::class)->create([
+            'experienceable_id' => $applicant->id,
+            'experienceable_type' => 'applicant',
+        ]);
+
+        $applicant->experiences_award()->save($experience);
+
+        $experienceSkill = factory(ExperienceSkill::class)->create([
+            'experience_id' => $experience->id,
+            'experience_type' => 'experience_award',
+        ]);
+
+        $experienceAward = $applicant->experiences_award()->first();
+        $experienceAward->experience_skills()->save($experienceSkill);
+
+        $skill = Skill::find($experienceSkill->skill_id);
+        $experienceAward->skills()->sync($skill->id);
+
+        // Create a new job application.
+        $application = factory(JobApplication::class)->make();
+        $applicant->job_applications()->save($application);
+
+        $application->saveProfileSnapshotTimeline();
+
+        $this->collectionConsistsOfCopy($applicant->experiences_award->first(), $application->fresh()->experiences_award);
+
+        // Ensure copies are linked to each other just like originals are.
+        $experienceAwardCopy = $application->experiences_award()->first();
+        $experienceSkillCopy = $experienceAwardCopy->experience_skills()->first();
+        $skillCopy = $experienceAwardCopy->skills()->first();
+
+        $this->assertEquals($application->id, $experienceAwardCopy->experienceable_id);
+        $this->assertEquals($experienceAwardCopy->experienceable_type, 'application');
+        $this->assertEquals($experienceAwardCopy->id, $experienceSkillCopy->id);
+        $this->assertEquals($experienceSkillCopy->skill_id, $skillCopy->id);
+    }
+
+    /**
+     * Ensure that saveProfileSnapshot doesn't cause many copies to be created if
+     * its called multiple times, and that it deletes copies that no longer match an original.
+     *
+     * @return void
+     */
+    public function testSaveProfileSnapshotTimelineResetsEachTime(): void
+    {
+        $applicant = factory(Applicant::class)->create();
+        $experience = factory(ExperienceAward::class)->create([
+            'experienceable_id' => $applicant->id,
+            'experienceable_type' => 'applicant',
+        ]);
+
+        $applicant->experiences_award()->save($experience);
+
+        $experienceSkill = factory(ExperienceSkill::class)->create([
+            'experience_id' => $experience->id,
+            'experience_type' => 'experience_award',
+        ]);
+
+        $experienceAward = $applicant->experiences_award()->first();
+        $experienceAward->experience_skills()->save($experienceSkill);
+
+        $skill = Skill::find($experienceSkill->skill_id);
+        $experienceAward->skills()->sync($skill->id);
```

Comment:
The ->skills attribute on ExperienceAward/Work/etc was intended to be a shortcut to accessing the skills associated with its ExperienceSkill. It wasn't be meant to use to create new relationships -  that should only be done by accessing the experience_skills relationship. After saving $experienceSkill to $experienceAward, $experienceAward->skills will already include $experienceSkill->skill_id.

In fact, in hindsight, this "shortcut" relationship is more confusing than useful. I'll just remove it and remove the reference to it here.